specification/kubernetesconfiguration/resource-manager/Microsoft.KubernetesConfiguration/preview/2021-05-01-preview/kubernetesconfiguration.json

```diff
@@ -0,0 +1,566 @@
+              "type": "string",
+              "default": "default"
+            },
+            "operatorInstanceName": {
+              "description": "Instance name of the operator - identifying the specific configuration.",
+              "type": "string"
+            },
+            "operatorType": {
+              "description": "Type of the operator",
+              "$ref": "#/definitions/OperatorTypeDefinition"
+            },
+            "operatorParams": {
+              "description": "Any Parameters for the Operator instance in string format.",
+              "type": "string"
+            },
+            "configurationProtectedSettings": {
+              "description": "Name-value pairs of protected configuration settings for the configuration",
+              "type": "object",
+              "$ref": "#/definitions/ConfigurationProtectedSettings"
+            },
+            "operatorScope": {
+              "description": "Scope at which the operator will be installed.",
+              "$ref": "#/definitions/OperatorScopeDefinition"
+            },
+            "repositoryPublicKey": {
+              "description": "Public Key associated with this SourceControl configuration (either generated within the cluster or provided by the user).",
+              "type": "string",
+              "readOnly": true
+            },
+            "sshKnownHostsContents": {
+              "description": "Base64-encoded known_hosts contents containing public SSH keys required to access private Git instances",
+              "type": "string"
+            },
+            "enableHelmOperator": {
+              "description": "Option to enable Helm Operator for this git configuration.",
+              "type": "boolean"
+            },
+            "helmOperatorProperties": {
+              "description": "Properties for Helm operator.",
+              "type": "object",
+              "$ref": "#/definitions/HelmOperatorProperties"
+            },
+            "provisioningState": {
+              "type": "string",
+              "description": "The provisioning state of the resource provider.",
+              "readOnly": true,
+              "x-ms-enum": {
+                "modelAsString": true,
+                "name": "ProvisioningStateType"
+              },
+              "enum": [
+                "Accepted",
+                "Deleting",
+                "Running",
+                "Succeeded",
+                "Failed"
+              ]
+            },
+            "complianceStatus": {
+              "type": "object",
+              "description": "Compliance Status of the Configuration",
+              "readOnly": true,
+              "$ref": "#/definitions/ComplianceStatus"
+            }
+          }
+        },
+        "systemData": {
+          "description": "Top level metadata https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/common-api-contracts.md#system-metadata-for-all-azure-resources",
```

Comment:
this should be readonly

Reply:
In CommonTypes.json, where systemData is defined, it is already marked as readOnly.